services/etcd.go

```diff
@@ -479,28 +479,41 @@ func RestoreEtcdSnapshot(ctx context.Context, etcdHost *hosts.Host, prsMap map[s
 		// printing the restore container's logs
 		return fmt.Errorf("Failed to run etcd restore container, exit status is: %d, container logs: %s", status, containerLog)
 	}
-	return docker.RemoveContainer(ctx, etcdHost.DClient, etcdHost.Address, EtcdRestoreContainerName)
+	if err := docker.RemoveContainer(ctx, etcdHost.DClient, etcdHost.Address, EtcdRestoreContainerName); err != nil {
+		return err
+	}
+	return RunEtcdSnapshotRemove(ctx, etcdHost, prsMap, etcdRestoreImage, snapshotName, true, es)
 }
 
-func RunEtcdSnapshotRemove(ctx context.Context, etcdHost *hosts.Host, prsMap map[string]v3.PrivateRegistry, etcdSnapshotImage string, name string, once bool, es v3.ETCDService) error {
+func RunEtcdSnapshotRemove(ctx context.Context, etcdHost *hosts.Host, prsMap map[string]v3.PrivateRegistry, etcdSnapshotImage string, name string, cleanup bool, es v3.ETCDService) error {
 	log.Infof(ctx, "[etcd] Removing snapshot [%s] from host [%s]", name, etcdHost.Address)
-	fullPath := fmt.Sprintf("/backup/%s{,.%s}", name, EtcdSnapshotCompressedExtension)
+
+	compressedPath := fmt.Sprintf("/backup/%s.%s", name, EtcdSnapshotCompressedExtension)
+	uncompressedPath := fmt.Sprintf("/backup/%s", name)
 	// Make sure we have a safe path to remove
-	safePath, err := filepath.Match("/backup/*", fullPath)
-	if err != nil {
-		return fmt.Errorf("failed to validate snapshot name: %v", err)
-	}
-	if !safePath {
-		return fmt.Errorf("malformed snapshot path: %v", fullPath)
+	for _, p := range []string{compressedPath, uncompressedPath} {
+		if safePath, err := filepath.Match("/backup/*", p); err != nil || !safePath {
+			return fmt.Errorf("invalid or malformed snapshot path [%s]: %v", p, err)
+		}
 	}
 
 	imageCfg := &container.Config{
-		Cmd: []string{
-			"sh", "-c", fmt.Sprintf("rm -f %s", fullPath),
-		},
 		Image: etcdSnapshotImage,
 		Env:   es.ExtraEnv,
 	}
+	if cleanup {
```

Comment:
can we rename cleanup to something that indicates its logic? Currently if cleanup is false, we remove everything